src/server.c

```diff
@@ -405,14 +405,24 @@ int dictExpandAllowed(size_t moreMem, double usedRatio) {
  * In non-cluster mode, we don't need this list as there is only one dictionary per DB. */
 void dictRehashingStarted(dict *d) {
     if (!server.cluster_enabled || !server.activerehashing) return;
+    server.db[0].sub_dict[DB_MAIN].bucket_count += DICTHT_SIZE(d->ht_size_exp[1]); /* Started rehashing (Add the new ht size) */
```

Comment:
i'm not too happy about the use of DICTHT_SIZE and ht_size_exp[1] (in contrast to dictBuckets).
seems to dive too deep into dict internals.
it's not a lot of places, so it'll be easy to clean in the future when things change or we have a better idea though.